lua/conform/formatters/prettierd.lua

```diff
@@ -1,5 +1,55 @@
 local fs = require("conform.fs")
 local util = require("conform.util")
+
+local config_file_names = {
+  -- https://prettier.io/docs/en/configuration.html
+  ".prettierrc",
+  ".prettierrc.json",
+  ".prettierrc.yml",
+  ".prettierrc.yaml",
+  ".prettierrc.json5",
+  ".prettierrc.js",
+  ".prettierrc.cjs",
+  ".prettierrc.mjs",
+  ".prettierrc.toml",
+  "prettier.config.js",
+  "prettier.config.cjs",
+  "prettier.config.mjs",
+}
+
+local function read_json(file)
+  local f = io.open(file, "r")
+
+  if f then
+    local file_content = f:read("*all") -- Read entire file contents
+    f:close()
+
+    return vim.fn.json_decode(file_content)
+  else
+    print("Error: Unable to open file " .. file)
```

Comment:
We have a logger for things like this
```suggestion
    log.warn("Unable to open file %s", file)
```

Reply:
I feel it's more applicable to throw an exception here. I'll check it.

Reply:
I disagree. There could be some people that use json5 for their package.json file (uncommon, but I've seen it -_-). That would cause this formatter to fail and spit out an error every time. It would be better to instead fall back to a best-effort guess using the other marker files as that will work just fine in most cases.
I could be convinced that it should log an error instead of a warning though, just to make sure it's visible by default in `:ConformInfo`

Reply:
Oh.. right. Then I think a warning should be fine until we are able to read json5.